### DISCUSSION TITLE
Possible Bug Found: Typo and Event Handling Inconsistency

### DISCUSSION THREAD

**Discussion by lukaszkowalik2** [fabricjs/fabric.js#9815] [General]
Hey there,

I stumbled upon a couple of quirks while experimenting with the `canvas.on()` method in Fabric.js, and I thought I'd give you a heads up:

1. Typo Alert:
When using `canvas.on()` with "object:modified" events, the IntelliSense in VSCode only shows `{target}` as available parameters. However, it should include `{e, target, transform, action}` for a more comprehensive event handling experience.

2. Event Removal Confusion:
Here's a peculiar thing I noticed when attempting to remove events like "mouse:move" from the canvas using `canvas.off()`. Even if I try to remove an event without specifying a function, like `canvas.off("mouse:move")`, it throws an error saying no event was provided for removal. However, despite the error message, the "mouse:move" function still gets removed successfully.

I've crafted a small example to demonstrate these issues, but unfortunately, it doesn't work correctly in the provided editor due to some strange problem with downloading libraries.

[Example Link](https://codesandbox.io/p/devbox/bug-forked-kk9k2v?file=%2Fsrc%2FApp.tsx%3A45%2C24)

I've been itching to start contributing to Fabric.js, and while I'm not a TypeScript expert, I'd love to pitch in and help out where I can. Let me know if there's anything I can do!

**Comment by asturur** [comment]
i may be wrong but in your example there is no way to trigger event removal

**Reply by lukaszkowalik2** [reply]
You're correct that there isn't a manual way to remove the event. However, the event removal is indeed handled in the cleanup function within the useEffect hook. I added button to remove this event.

You can check out the example [here](https://codesandbox.io/p/devbox/bug-forked-kk9k2v?file=%2Fsrc%2FApp.tsx%3A2%2C37).

**Reply by asturur** [reply]
i have see the useEffect, but it wasn't unclear how i was supposed to check the demo

**Reply by asturur** [reply]
So i click the button but there are no errors.

**Reply by lukaszkowalik2** [reply]
I apologize for the confusion. It seems I worded my previous comment incorrectly. There might indeed be a type error in the code. I believe there is a type error present.

**Reply by asturur** [reply]
i need to understand what doesn't work correctly and what are you seeing. Then maybe i can help, so far i m still in the dark :)
